dask/dataframe/reshape.py

```diff
@@ -358,13 +359,15 @@ def melt(
 
     from dask.dataframe.core import no_default
 
-    return frame.map_partitions(
-        M.melt,
-        meta=no_default,
-        id_vars=id_vars,
-        value_vars=value_vars,
-        var_name=var_name,
-        value_name=value_name,
-        col_level=col_level,
-        token="melt",
-    )
+    # let pandas do upcasting as needed during melt
+    with config.set({"dataframe.convert_string": False}):
```

Comment:
Not a big deal, or meant as a blocking comment, but I normally see 

```python
import dask
...
dask.config.get(...)
dask.config.set(...)
```

Reply:
Sure, that's an easy change.